source/os/vss-command_line.ads

```diff
@@ -0,0 +1,73 @@
+--
+--  Copyright (C) 2022, AdaCore
+--
+--  SPDX-License-Identifier: Apache-2.0
+--
+
+--  This package provides access to command line arguments as parsed options.
+
+with Ada.Containers.Vectors;
+
+with VSS.Strings;
+with VSS.String_Vectors;
+
+package VSS.Command_Line is
+
+   type Name_Value is record
+      Name  : VSS.Strings.Virtual_String;
+      Value : VSS.Strings.Virtual_String;
+   end record;
+
+   package Name_Value_Vectors is
+      new Ada.Containers.Vectors (Positive, Name_Value);
+
+   type Abstract_Option is abstract tagged record
+      Description : VSS.Strings.Virtual_String;
+   end record;
+
+   type Positional_Option is new Abstract_Option with record
+      Name : VSS.Strings.Virtual_String;
+   end record;
+
+   type Named_Option is abstract new Abstract_Option with record
+      Short_Name : VSS.Strings.Virtual_String;
+      Long_Name  : VSS.Strings.Virtual_String;
+   end record;
+
+   type Binary_Option is new Named_Option with null record;
+
+   type Value_Option is new Named_Option with record
+      Value_Name : VSS.Strings.Virtual_String;
+   end record;
+
+   type Name_Value_Option is new Named_Option with record
+      Name_Name  : VSS.Strings.Virtual_String;
+      Value_Name : VSS.Strings.Virtual_String;
+   end record;
+
+   procedure Add_Option (Option : Abstract_Option'Class);
+
+   procedure Process;
+   --  Parse command line. If error is found then report it and terminate
+   --  application.
```

Comment:
Terminating application looks as a bad idea to me. I prefer this procedure to return a parsing status and let the user react by her self.

Reply:
It is part of contract for this subprogram, it makes application code cleaner. 

Subprogram to do parsing with status may be added later if necessary.